olx.py

```diff
@@ -0,0 +1,342 @@
+                                if char.isdigit():
+                                    floor_number += char
+                            output["floor"] = int(floor_number)
+                        elif "rooms" in value:
+                            translate = {"one": 1, "two": 2, "three": 3, "four": 4}
+                            output["rooms"] = translate[value.split('"')[3]]
+                        elif "builttype" in value:
+                            output["builttype"] = value.split('"')[3]
+                        elif "furniture" in value:
+                            if value.split('"')[3] == "yes":
+                                output["furniture"] = True
+                            else:
+                                output["furniture"] = False
+    return output
+
+
+def parse_available_offers(markup):
+    html_parser = BeautifulSoup(markup, "html.parser")
+    not_found = html_parser.find(class_="emptynew")
+    if not_found is not None:
+        print("No offers found")
+        return
+    offers = html_parser.find_all(class_='offer')
+    parsed_offers = [parse_offer_url(str(offer)) for offer in offers if offer][OFFERS_FEATURED_PER_PAGE:]
+    return parsed_offers
+
+
+def parse_offer(markup, url):
+    html_parser = BeautifulSoup(markup, "html.parser")
+    offer_content = html_parser.find(class_='offerbody')
+    parsed_title = get_title(str(offer_content))
+    parsed_price = get_price(str(offer_content))
+    parsed_imgs = get_img_url(str(offer_content))
+    parsed_date = get_date_added(str(offer_content))
+    parsed_yardage = get_yardage(str(offer_content))
+    description = parse_description(str(offer_content))
+    offer_content = html_parser.body
+    offer_data = parse_flat_data(str(offer_content))
+    keys = list(offer_data.keys())
+    values = list(offer_data.values())
+    return {
+        "title": parsed_title,
+        "price": parsed_price,
+        "yardage": parsed_yardage,
+        keys[0]: values[0],
+        keys[1]: values[1],
+        keys[2]: values[2],
+        keys[3]: values[3],
+        keys[4]: values[4],
+        "description": description,
+        "url": url,
+        "date": parsed_date,
+        "images": parsed_imgs
+    }
+
+
+def get_category(main_category, subcategory, detail_category, region, *args):
+    parsed_content = []
+    page = 1
+    page_attr = None
+    while True:
+        url = get_url(page_attr, main_category, subcategory, detail_category, region, *args)
+        print(url)
+        response = get_content_for_url(url)
+        if response.status_code > 300:
+            break
+        print("Loaded page {} of offers".format(page))
+        offers = parse_available_offers(response.content)
+        if offers is None:
+            break
+        parsed_content.append(offers)
+        page += 1
+        page_attr = "page={}".format(page)
+    parsed_content = list(flatten(parsed_content))
+    print("Loaded " + str(len(parsed_content)) + " offers")
```

Comment:
we should remove prints and maybe use logging[1]


[1] - https://docs.python.org/3/howto/logging.html